Description: Fix OpenSSL 1.1 FTBFS
 OpenSSL 1.1 has changed the HMAC_* API slightly, this patch is needed to
 enable compilation of autobahn. It has been tested to work.
 .
Author: Konstantinos Margaritis <markos@debian.org>

--- autobahn-cpp-0.2.1.orig/autobahn/wamp_auth_utils.hpp
+++ autobahn-cpp-0.2.1/autobahn/wamp_auth_utils.hpp
@@ -159,12 +159,13 @@ inline std::string compute_wcs(
     unsigned int len = 32;
     unsigned char hash[32];
 
-    HMAC_CTX hmac;
-    HMAC_CTX_init(&hmac);
-    HMAC_Init_ex(&hmac, key.data(), key.length(), EVP_sha256(), NULL);
-    HMAC_Update(&hmac, ( unsigned char* ) challenge.data(), challenge.length());
-    HMAC_Final(&hmac, hash, &len);
-    HMAC_CTX_cleanup(&hmac);
+    HMAC_CTX *hmac = HMAC_CTX_new();
+    if (!hmac)
+        return "";
+    HMAC_Init_ex(hmac, key.data(), key.length(), EVP_sha256(), NULL);
+    HMAC_Update(hmac, ( unsigned char* ) challenge.data(), challenge.length());
+    HMAC_Final(hmac, hash, &len);
+    HMAC_CTX_free(hmac);
 
 
     std::string str_out;
